colossalai/booster/plugin/hybrid_parallel_plugin.py

```diff
@@ -1146,7 +1205,14 @@ def configure(
                         tp_process_group=self.tp_group,
                     )
             else:
-                if self.dp_size == 1:
+                # Here we bind the ZeRO group with sp group when user enable both ZeRO and all_to_all sp.
+                if self.enable_sequence_parallelism and self.sequence_parallelism_mode == "all_to_all":
+                    self.zero_dp_size = self.sp_size * self.dp_size
+                    self.zero_dp_group = self.pg_mesh.create_group_along_axis([DP_AXIS, SP_AXIS])
+                else:
+                    self.zero_dp_size = self.dp_size
+                    self.zero_dp_group = self.dp_group
```

Comment:
Can we keep the same logic as using ddp?

Reply:
fixed